src/lib.rs

```diff
@@ -359,11 +362,84 @@ pub fn init_and_run(options: Options) -> ! {
                     session = Session::new();
                     session.set_autorun_delay(current_autorun_delay);
 
-                    for stmt in project.stmts {
-                        session.push_prog_stmt(time, stmt);
+                    project_status.path = None;
+                    project_status.changed_since_last_save = false;
+
+                    change_window_title(&window, &project_status);
+                }
+
+                if let Some(save_path) = menu_status.save_path {
+                    log::info!("Saving project at {}", save_path);
+
+                    let stmts = session.stmts().to_vec();
+                    let project = project::Project { version: 1, stmts };
+
+                    match project::save(&save_path, project) {
+                        Ok(_) => {
+                            project_status.save(&save_path);
+
+                            change_window_title(&window, &project_status);
+                        }
+                        Err(err) => {
+                            log::error!("{}", err);
+                            project_status.error = Some(err);
+                        }
                     }
+                }
+
+                if let Some(open_path) = menu_status.open_path {
+                    log::info!("Opening new project at {}", open_path);
+
+                    match project::open(&open_path) {
+                        Ok(project) => {
+                            scene_meshes.clear();
+
+                            for (_, gpu_mesh_handle) in scene_gpu_mesh_handles.drain() {
+                                renderer.remove_scene_mesh(gpu_mesh_handle);
+                            }
+
+                            scene_bounding_box = BoundingBox::union(
+                                scene_meshes.values().map(|mesh| mesh.bounding_box()),
+                            )
+                            .unwrap_or_else(BoundingBox::unit);
+
+                            ground_plane_mesh = compute_ground_plane_mesh(&scene_bounding_box);
+                            ground_plane_mesh_bounding_box = ground_plane_mesh.bounding_box();
+                            renderer.remove_scene_mesh(
+                                ground_plane_gpu_mesh_handle
+                                    .take()
+                                    .expect("Ground plane must always be present"),
+                            );
+                            ground_plane_gpu_mesh_handle = Some(
+                                renderer
+                                    .add_scene_mesh(&GpuMesh::from_mesh(&ground_plane_mesh), true)
+                                    .expect("Failed to add ground plane mesh"),
+                            );
```

Comment:
This probably shouldn't be here. The recomputing should happen when new results are produced by the interpreter. It will either happen automatically (when autorun is enabled), or manually. Either way, we probably shouldn't do it here, before the geometries in the scene actually change.

This is also true for new project. (I am sorry mislead you here in your previous PR)

Reply:
Hmm, maybe we actually do want to keep this. Looks like a UX decision, so perhaps @janper should decide that. Do we want to clear the scene immediately when opening/newing a project? Or do we want to wait for the interpreter to run the new (potentially empty) program?

Reply:
I have kept this as is until you two decide what to do with it.

Reply:
@janper Didn't reply. Let's keep it.

Reply:
Sorry, I didn't get any notification.
Also I don't know what does this code do.

I believe when a project is loaded, the scene should be wiped and set to default state. The geometry should be generated only when the pipeline is executed (either by a button or automatically).

Reply:
Alright. I'll create a task to ensure this